awkward/array/jagged.py

```diff
@@ -24,6 +24,12 @@ class JaggedArray(awkward.array.base.AwkwardArrayWithContent):
 
     @classmethod
     def offsetsaliased(cls, starts, stops):
+        # short circuit for arrow-backed arrays
+        if starts.base is not None and stops.base is not None:
+            if starts.base.base is not None and stops.base.base is not None:
+                if (isinstance(starts.base.base, awkward.pyarrow.lib.Buffer) and
+                    isinstance(stops.base.base, awkward.pyarrow.lib.Buffer)):
+                    return (starts.base.base.address == starts.base.base.address)
```

Comment:
In uproot, I avoid dependencies on projects that I nevertheless want to check for by checking module and class name: https://github.com/scikit-hep/uproot/blob/163bf0ab0a5b9d16e7aee61b8ab19e0b0412a83d/uproot/tree.py#L119 . The justification is that import names are just strings in a namespace anyway.

Maybe this is a general technique that we would use on any `from*` that gives us a buffer that isn't a NumPy array. In that case, we can just check `if not isinstance(starts.base.base, numpy.ndarray)` and apply the same logic to all such cases. That said, I hope the new Awkward is ready before we hit any other such cases.

Reply:
I don't think this check inside `offsetsaliased` is necessary.  Its sufficient to sanitize on construction no?

Reply:
I, too, prefer sanitizing on construction, because then you don't potentially break this code for other purposes. `offsetsaliased` is used a lot, and it's fail-fast: if anything is amiss, it assumes that the offsets are not aliased, because that's the safe-though-maybe-not-fast option.